compiler/vm.nim

```diff
@@ -242,6 +276,15 @@ proc writeField(n: var PNode, x: TFullReg) =
   of rkNodeAddr: n = x.nodeAddr[]
 
 proc putIntoReg(dest: var TFullReg; n: PNode) =
+  if n.kind == nkIntLit:
```

Comment:
Put this inside the ``case n.kind`` handling by splitting the case like so

```nim

of nkCharLit..pred(nkIntLit), succ(nkIntLit)..nkUInt64Lit:
   ...
of nkIntLit:
  ...
```

Reply:
done; but simplified further as: `of {nkCharLit..nkUInt64Lit} - {nkIntLit}`; there are a few places in stdlib that could use this form, which is simpler than pred/succ